.github/workflows/linters.yml

```diff
@@ -0,0 +1,32 @@
+name: Format code using black and isort
+# modified from https://towardsdatascience.com/black-with-git-hub-actions-4ffc5c61b5fe
+# caching https://stackoverflow.com/questions/59127258/how-can-i-use-pip-cache-in-github-actions
+# black integration https://black.readthedocs.io/en/stable/integrations/github_actions.html
+# isort integration https://pycqa.github.io/isort/docs/configuration/github_action.html
+
+
+
+on:
+  push:
+    branches: [ "develop", "issue23-precommithooks" ]
```

Comment:
Automated code cleanup on: pull_request would, I think, be redundant as it's already been cleaned up on any push (any commit to the repo).  Besides pull_request triggers when a PR is opened, closed, or reopened.  A push triggers when the PR is accepted and the code is committed to the branch.  So I'm not sure it captures the moment we want to - ie: code being added.

We can remove the branches filter entirely or use branches : ** (to be explicit) and it will lint all branches.

We can also exclude specific branches, so;

on:
  push:
    branches-ignore:  ['master']  

That should trigger for a push to every branch but the master branch

I'll change this to all but the master branch for now.